core/domain/story_validators_test.py

```diff
@@ -0,0 +1,1125 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for core.domain.story_validators."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+
+from constants import constants
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import learner_progress_services
+from core.domain import prod_validation_jobs_one_off
+from core.domain import rights_manager
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.domain import user_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+import python_utils
+
+datastore_services = models.Registry.import_datastore_services()
+
+USER_EMAIL = 'useremail@example.com'
+USER_NAME = 'username'
+CURRENT_DATETIME = datetime.datetime.utcnow()
+
+(
+    base_models, exp_models,
+    story_models, topic_models, user_models
+) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.exploration,
+    models.NAMES.story, models.NAMES.topic, models.NAMES.user
+])
+
+
+class StoryModelValidatorTests(test_utils.AuditJobsTestBase):
+
+    def setUp(self):
+        super(StoryModelValidatorTests, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        explorations = [exp_domain.Exploration.create_default_exploration(
+            '%s' % i,
+            title='title %d' % i,
+            category='category',
+        ) for i in python_utils.RANGE(6)]
```

Comment:
Did you accidentally delete the line `correctness_feedback_enabled=True` from this declaration? I think that's only causing the backend tests to fail. Please use the old code taken from the `prod_validators_test` file (written below):
```python
        explorations = [self.save_new_valid_exploration(
            '%s' % i,
            self.owner_id,
            title='title %d' % i,
            end_state_name='End State',
            correctness_feedback_enabled=True
        ) for i in python_utils.RANGE(4)]
```

Reply:
Hi @sajalasati, I did fix that part. But I am getting another error. Here is what one of them looks like:
```
ERROR: test_completed_node_missing_in_story_node_ids (core.domain.story_validators_test.StoryProgressModelValidatorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/purvi/Documents/opensource/oppia/core/domain/story_validators_test.py", line 1091, in test_completed_node_missing_in_story_node_ids
    expected_output, sort=False, literal_eval=False)
  File "/home/purvi/Documents/opensource/oppia/core/tests/test_utils.py", line 2792, in run_job_and_check_output
    self.process_and_flush_pending_mapreduce_tasks()
  File "/home/purvi/Documents/opensource/oppia/core/tests/test_utils.py", line 2654, in process_and_flush_pending_mapreduce_tasks
    self._execute_mapreduce_tasks(tasks)
  File "/home/purvi/Documents/opensource/oppia/core/tests/test_utils.py", line 2634, in _execute_mapreduce_tasks
    'MapReduce task to URL %s failed' % task.url)
RuntimeError: MapReduce task to URL /mapreduce/worker/worker_callback/1557734249399testbed-request-id-hash-1 failed
```

Do you know what might be causing this? I am facing some problems debugging the code in vscode as well.

Reply:
@purvimisal This error is due to some map reduce task failing, and at the moment I cannot really figure out what might be causing it to fail. What about other tests in this class, which other tests are also failing?
I suggest you push the code first, and then it would be easier to debug given the exact code and log output of all the tests.